chainforge/react-server/src/store.js

```diff
@@ -112,6 +112,48 @@ export const initLLMProviders = [
     base_model: "azure-openai",
     temp: 1.0,
   },
+  {
+    name: "Bedrock Claude",
+    emoji: "👨‍🏫",
+    model: "anthropic.claude-v2:1",
+    base_model: "anthropic.claude",
+    temp: 0.9,
+  },
+  {
+    name: "Bedrock Jurassic 2",
+    emoji: "🦖",
+    model: "ai21.j2-ultra",
+    base_model: "ai21.j2",
+    temp: 0.9,
+  },
+  {
+    name: "Bedrock Titan",
+    emoji: "🏛️",
+    model: "amazon.titan-tg1-large",
+    base_model: "amazon.titan",
+    temp: 0.9,
+  },
+  {
+    name: "Bedrock Command Text 14",
+    emoji: "📚",
+    model: "cohere.command-text-v14",
+    base_model: "cohere.command",
+    temp: 0.9,
+  },
+  {
+    name: "Bedrock Mistral",
+    emoji: "💨",
+    model: "mistral.mistral-7b-instruct-v0:2",
+    base_model: "mistral.mistral",
+    temp: 0.9,
+  },
+  {
+    name: "Bedrock Meta",
+    emoji: "🦙",
+    model: "meta.llama2-13b-chat-v1",
+    base_model: "meta.llama2",
+    temp: 0.9,
+  },
```

Comment:
In general, we don't want to clutter the providers list. The model providers list is generally about Providers rather than individual models (yes, I know I break this once with GPT-4 and GPT-3.5 as separate items, but all other providers only take up a single space). As Bedrock is a single provider, it should have take up only a single space in the list.

However, I see the potential problem you are facing here. In ChainForge, the usual way to add a provider is pick a "default" model (e.g. Claude-3) and set the "base_model" for all of these to just be "bedrock." Then you'd have a single settings window, for **all** Bedrock models, with a "model" dropdown to change the model. 

However, in your case, all of these models may have custom APIs and may be incompatible with a single settings window. If that is true, then, you will have to either figure out how a single settings window could work for all of them (is there significant enough overlap that it's fine?), or, if that is not possible, then you need to change the Mantine context <Menu /> for providers to use a submenu: https://icflorescu.github.io/mantine-contextmenu-v6/examples/submenus

So, if you'd prefer the latter case, the new Menu would have a Bedrock item with an -> arrow to a submenu with these Bedrock models. It shouldn't be that hard to implement, and the rest of the codebase may benefit from this change (in particular I would make OpenAI its own item with > to GPT3.5 and GPT4, for symmetry's sake).

Reply:
As you rightfully point out I encountered the issue that Bedrock is just a model hosting provider and not a model provider. As model hosting provider it gives you the ability to use a single, unified API to invoke models, and moves the providers/model specificities into the content of the "body" that is sent into that API. You could see Bedrock approach similar to HuggingFace [Inference API](https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task), but where HF chose a minimum common denominator approach, in Bedrock the same API (InvokeModel) is just a pass through to the underlying model being used and therefore you can use any of the model specific parameters (see Ai21 models for example). 
With all this said, I'll go for the sub-menu approach (thanks for suggesting that) to unclutter the UI but retain the model specific parameters. I'll also apply it to OpenAI for consistency.

Reply:
Since we are talking about models, I thought about refactoring `ModelSettingsSchemas.js` and split each provider in a separate file as this is growing pretty big. What do you think?

Reply:
![image](https://github.com/ianarawjo/ChainForge/assets/38036163/1071df47-6d27-47f0-83f0-5961acff5949)

This is how it looks with sub-menus (I am not using https://icflorescu.github.io/mantine-contextmenu-v6/examples/submenus)

Reply:
Is there a reason not to use mantine context menu? It looks like the alignments are messed up and the icon being used for the arrow is not ideal. I would prefer to use context menu to be consistent.

Reply:
> Since we are talking about models, I thought about refactoring `ModelSettingsSchemas.js` and split each provider in a separate file as this is growing pretty big. What do you think?

I agree with you here that we need to begin thinking of another solution. However, I have a pending Typescript branch that I need to merge. Refactoring would make that merge even harder than it will be currently. So, maybe could you raise a GitHub issue about this? Then we can get to it later. 

Thank you so much for your work on this!

Reply:
<img width="623" alt="image" src="https://github.com/ianarawjo/ChainForge/assets/38036163/2496b079-3314-49e7-9731-e36e0e7cd9c4">

Reply:
This is the new version with context menu.

Reply:
This is amazing!!!!! 🤩 Is there a way to add icons even for the > ones? Like a rock for BedRock, hug emoji for HF, etc? Also feel free to put Bedrock right below HuggingFace, given the work you've done on this.